### ISSUE TITLE
File Conversions Not Created with S3

### ISSUE THREAD

**Issue by bakerstreetsystems** [spatie/laravel-medialibrary#325]
I'm using Spatie/Laravel-Medialibrary to handle file attachments to my Product model. I'd like to have all product images upload to Amazon S3 and then have each image converted to different sizes automatically (i.e. "thumb", "small", "medium", "large").

Everything works great when working with the local filesystem. The package creates the conversions and I'm able to access them with my app. The problem comes when I try to change the package's settings to use Amazon S3 instead of the local file storage.

So in my `ImageController.php`, I have the following line:

`$product->addMedia(public_path() . '/temp/products/'.$product->id)->toCollection('images', 's3');
`
S3 config in `config/filesystems.php`

```
    's3' => [
            'driver' => 's3',
            'key' => env('AMAZON_S3_ID'),
            'secret' => env('AMAZON_S3_KEY'),
            'region' => env('AMAZON_S3_REGION'),
            'bucket' => env('AMAZON_S3_BUCKET'),
            'visibility' => 'public'
        ],
```

and the conversion settings located in my `Product.php` class:

```
    public function registerMediaConversions()
    {
        $this->addMediaConversion('thumb')
             ->setManipulations(['w' => 160, 'h' => 120])
             ->performOnCollections('images');

        $this->addMediaConversion('small')
             ->setManipulations(['w' => 280, 'h' => 210])
             ->performOnCollections('images');

        $this->addMediaConversion('medium')
             ->setManipulations(['w' => 400, 'h' => 300])
             ->performOnCollections('images');

        $this->addMediaConversion('large')
             ->setManipulations(['w' => 640, 'h' => 480])
             ->performOnCollections('images');
    }
```

As I mentioned above, everything works great if I use the local file system, but as soon as I change it to use Amazon S3, the file conversions are not created. The original file is successfully uploaded to Amazon, but the conversions are not there. Any advice?

**Comment by nicolasbeauvais** [comment]
Hi @bakerstreetsystems,

can you please check this recent similar issue #321, and make all the checks mentioned there ?

Does `php artisan medialibrary:regenerate` create the conversions ?
Are you using a queue ?
...

Thanks !

**Comment by bakerstreetsystems** [comment]
Thanks for replying. I've looked through the recent similar issue you suggested ([#321](https://github.com/spatie/laravel-medialibrary/issues/321)), and though both issues are related to file conversions, I believe issue 321 has to do with possible errors in the Laravel queue configuration whereas the issue I'm having trying to use conversions with S3 doesn't ever get to the queue step.

My queue driver up to this point has been 'sync', and as I mentioned in my original post, everything works perfectly when the filesystem is set to 'local' or 'media'. But when I switch it to S3, only the original image gets uploaded to S3 and I don't think laravel-medialibrary even attempts to execute the conversions.

To double check this, I changed my queue driver to 'database' so I could see if jobs were even being queued. With the filesystem set to 'local' or 'media', the conversion job was successfully queued and I could see the record in the 'jobs' table in my database. With the 's3' filesystem, no job was queued.

I then tried adding `->nonQueued()` to each `addMediaConversion` in the `registerMediaConversions` method in my `Product` model. Again, it works with the 'local' and 'media' filesystems, but not with the 's3' filesystem.

Anything else I should try?

**Comment by nicolasbeauvais** [comment]
I suggested the issue because queuing (if activated) is the biggest difference between uploading a file and it's conversions, and also one of the only cases when you will not see any errors.

What I would suggest to you is either activating logs on your s3 bucket (this can add some costs to your bucket if you make a lot of operation on it). But first, maybe add some debug lines to the media library core to check when using the s3 driver that:

- the conversion of the file is called:
  - https://github.com/spatie/laravel-medialibrary/blob/master/src/FileManipulator.php#L23
  - https://github.com/spatie/laravel-medialibrary/blob/master/src/FileManipulator.php#L38
- the file is created:
  - https://github.com/spatie/laravel-medialibrary/blob/master/src/FileManipulator.php#L79
- the conversion file is sent to s3:
  - https://github.com/spatie/laravel-medialibrary/blob/master/src/Filesystem.php#L48

**Comment by boofman15** [comment]
Just wanted to throw something I found out regarding conversions on S3.  Some images I submitted were saving and getting converted fine, some were not.  Found out that images that have ".JPG" instead of ".jpg" were not saved.  Looks like some code in the ImageGenerators/FileTypes classes' convert function try to reference files with ".jpg" so it was not able to find my files with uppercase extensions.

I can make sure my extensions are lower case when saved to S3, but should uppercase extensions be handled in the converter as well?

**Comment by nicolasbeauvais** [comment]
@boofman15 Great catch ! I made a PR #370 to fix it

**Comment by freekmurze** [comment]
Thanks @nicolasbeauvais 

Fix is available in `v4.9.3`.

**Comment by nicolasbeauvais** [comment]
I'm not sure that this issue should be closed, #329 hasn't been fixed yet

**Comment by bakerstreetsystems** [comment]
I just tried using the latest version, and it appears to work great now! This issue may be closed.
